spark/src/main/java/org/opensearch/sql/spark/dispatcher/QueryIdProvider.java

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.spark.dispatcher;
+
+import org.opensearch.sql.spark.dispatcher.model.DispatchQueryRequest;
+
+/** Interface for extension point to specify queryId. Called when new query is executed. */
+public interface QueryIdProvider {
+  String getQueryId(DispatchQueryRequest dispatchQueryRequest);
```

Comment:
is this parameter sufficient for generating query id? or are we expecting implementation to use thread context to pass them

Reply:
It should be sufficient.